ios/engine/KMEI/KeymanEngine/Classes/Storage.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  Storage.swift
+//  KeymanEngine
+//
+//  Created by Gabriel Wong on 2017-11-21.
+//  Copyright © 2017 SIL International. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - Static members
+extension Storage {
+  /// Storage that is not shared with the other process (app or extension).
+  static let nonShared: Storage? = {
+    let paths = FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask)
+    if paths.isEmpty {
+      return nil
+    }
+    return Storage(baseURL: paths[0], userDefaults: UserDefaults.standard)
+  }()
+
+  /// Storage shared between the app and app extension.
+  /// - Important: `Manager.applicationGroupIdentifier` must be set before accessing this. Modifications to
+  ///   `Manager.applicationGroupIdentifier` will not be reflected here.
+  static let shared: Storage? = {
+    guard let groupID = Manager.applicationGroupIdentifier else {
+      return nil
```

Comment:
Had to temporarily remove logging during initialization of `Storage` since `Manager.init()` uses `Storage` and so we can't quite access `Manager.shared.kmLog()` here yet.

This can be fixed by moving `kmLog()` out of `Manager`. Hoping to do that after we set up infrastructure to pull in libraries (which we will need to do to unzip .kmp), so we can use a small logging library.